File: src/main/java/tn/esprit/devflow/courzelo/entity/EventRegistration.java
package tn.esprit.devflow.courzelo.entity;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.util.List;

@Data
@Document
@FieldDefaults(level= AccessLevel.PRIVATE)

public class EventRegistration {
    @Id
    String idreg;
    LocalDate registrationdate;
    Boolean confirmationstatus;
   @DBRef
   List<User> users;
}

File: Courzelo_Angular22/src/app/models/Event/EventRegistration.ts
import { Event } from "./event";

export interface EventRegistration {
    idreg: string;
    registrationDate: Date;
    confirmationStatus: boolean;
    events: Event[];
    user: User;
}

export interface User {
    id: string;
    username: string;
    email: string;
    // other user details
}